ui/src/modal.ts

```diff
@@ -1,7 +1,12 @@
 import { derived, get, writable } from "svelte/store";
 
+type OnHide = () => void;
+const doNothing = () => {
+  /**/
+};
+
 type ModalOverlay =
-  | { show: true; route: string }
+  | { show: true; route: string; onHide: OnHide }
```

Comment:
Wouldn’t it be more appropriate to call this `onClose`?

Reply:
Also thought so. However, we use the 'hide' terminology elsewhere.